### ISSUE TITLE
Unprivileged user mode broken in master branch

### ISSUE THREAD

**Issue by DrDaveD** [apptainer/singularity#1847]
## Version of Singularity:

Current master (3.0) branch

### Expected behavior

The exec -u option will work on an RHEL7 system with user namespaces enabled

### Actual behavior

ERROR  : Failed to write to GID map: Operation not permitted

### Steps to reproduce behavior

/usr/local/bin/singularity exec -u yourfavoritecontainer.sandbox bash

**Comment by cclerget** [comment]
@DrDaveD Can't reproduce on a CentOS 7.4.1708. What RHEL version it is ? Can you paste the debug output ?

**Comment by DrDaveD** [comment]
This was Scientific Linux 7.5.   Do you have a kernel also from CentOS 7.4 or are you using a container?  Here are the [instructions I used for enabling user namespaces](https://opensciencegrid.org/docs/worker-node/install-singularity/#enabling-singularity-via-oasis).

Here's the output with -vvv -d:
```
VERBOSE [U=3382,P=19655]   singularity_message_level()               Set messagelevel to: 5
DEBUG   [U=3382,P=19655]   init()                                    PIPE_EXEC_FD value: 7
VERBOSE [U=3382,P=19655]   init()                                    Container runtime
VERBOSE [U=3382,P=19655]   is_suid()                                 Check if we are running as setuid
DEBUG   [U=3382,P=19655]   init()                                    Read json configuration from pipe
DEBUG   [U=3382,P=19655]   init()                                    Set child signal mask
DEBUG   [U=3382,P=19655]   init()                                    Create socketpair for smaster communication channel
DEBUG   [U=3382,P=19655]   init()                                    Wait C and JSON runtime configuration from scontainer stage 1
DEBUG   [U=3382,P=19660]   set_parent_death_signal()                 Set parent death signal to 9
VERBOSE [U=3382,P=19660]   init()                                    Spawn scontainer stage 1
VERBOSE [U=3382,P=19660]   main()                        Execute scontainer stage 1
DEBUG   [U=3382,P=19660]   SContainer()                  Entering scontainer stage 1
DEBUG   [U=3382,P=19655]   init()                                    Receiving configuration from scontainer stage 1
DEBUG   [U=3382,P=19655]   init()                                    Wait completion of scontainer stage1
VERBOSE [U=3382,P=19655]   setup_userns()                            Create user namespace
DEBUG   [U=65534,P=19655]  setup_userns()                            Write deny to set group file
DEBUG   [U=65534,P=19655]  setup_userns()                            Write to GID map
DEBUG   [U=65534,P=19655]  setup_userns()                            Write line '3382 3382 1' to gid_map
ERROR   [U=65534,P=19655]  setup_userns()                            Failed to write to GID map: Operation not permitted
```

**Comment by cclerget** [comment]
I used a VM with `Linux localhost.localdomain 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux` kernel. The instructions looks good and uid/gid from debug output seems correct.

I'm confused, there is nothing fancy in user namespace code, debug output don't show SUID workflow. Does `unshare -u -r -m` work ?

**Comment by DrDaveD** [comment]
Yes the unshare command works, as does unprivileged singularity with 2.6.0.  My kernel is considerably newer 3.10.0-862.6.3.el7.x86_64, can you upgrade the kernel?  It's the EL7.5 kernel but it is the default security update kernel for all 7.x.  

I will also try to compare exactly what the different system calls are between 2.6.0 and the master branch.

**Comment by cclerget** [comment]
Works with upgraded kernel. I used go 1.10, can you try to with go 1.10 ?

**Comment by DrDaveD** [comment]
It didn't help.   There must be something else different about our environments.  Still checking.

**Comment by DrDaveD** [comment]
The difference was that I have an account where uid is not equal to gid.  Fixed in #1856.
